perf/ring_test.go

```diff
@@ -62,3 +64,44 @@ func makeRing(size, offset int) *ringReader {
 
 	return newRingReader(&meta, ring)
 }
+
+func TestPerfEventRing(t *testing.T) {
+	check := func(buffer, watermark int) {
+		ring, err := newPerfEventRing(0, buffer, watermark)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		size := len(ring.ringReader.ring)
+
+		// Ring size should be at least as big as buffer
+		if size < buffer {
+			t.Fatalf("ring size %d smaller than buffer %d", size, buffer)
+		}
+
+		// Ring size should be of the form 2^n pages (meta page has already been removed)
+		pages := float64(size) / float64(os.Getpagesize())
```

Comment:
Why not `size % os.Getpagesize() != 0`? I find the math.Trunc a bit hard to follow.

Reply:
Nice, that is clearer! Fixed :)